src/main/java/net/greatstart/dao/hibernate/InvestmentInterestDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package net.greatstart.dao.hibernate;
+
+import net.greatstart.dao.InvestmentInterestDao;
+import net.greatstart.model.InvestmentInterest;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class InvestmentInterestDaoImpl implements InvestmentInterestDao {
+    private SessionFactory sessionFactory;
+
+    @Autowired
+    public InvestmentInterestDaoImpl(SessionFactory sessionFactory) {
+
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override

```

Comment:
The next our step will be crossing to Spring DATA(CrudRepository or JpaRepository), why to do that?